Rakefile

```diff
@@ -0,0 +1,150 @@
+      puts "Barclamp #{barclamp} already forked. Skipping..."
+    else
+      puts "Forking barclamp #{barclamp}..."
+      client.fork("crowbar/#{barclamp}")
+    end
+  end
+end
+
+desc "Clones all required barclamps into the barclamps folder."
+task :clone => [:configure] do
+  client = Octokit::Client.new(:netrc => true)
+
+  Dir.chdir("barclamps") do
+    @all_barclamps.each do |barclamp|
+      if Dir.exist?(barclamp)
+        puts "Barclamp #{barclamp} already exists. Skipping..."
+      else
+        puts "Cloning barclamp #{barclamp}..."
+        Git.clone("git@github.com:#{client.login}/barclamp-#{barclamp}.git", barclamp)
+      end
+    end
+  end
+end
+
+desc "Add upstream remote to the crowbar organisation."
+task :add_upstream => [:configure] do
+  @all_barclamps.each do |barclamp|
+    Dir.chdir File.join("barclamps", barclamp) do
+      git = Git::Lib.new
+      unless git.remotes.include?("upstream")
+        puts "Adding remote upstream for #{barclamp}..."
+        git.remote_add("upstream", "git@github.com:crowbar/barclamp-#{barclamp}.git")
+      end
+    end
+  end
+end
+
+desc "Add upstream remote to the SUSE-Cloud organisation."
+task :add_susecloud => [:configure] do
+  @all_barclamps.each do |barclamp|
+    Dir.chdir File.join("barclamps", barclamp) do
+      git = Git::Lib.new
+      unless git.remotes.include?("suse-cloud")
+        puts "Adding remote suse-cloud for #{barclamp}..."
+        git.remote_add("suse-cloud", "git@github.com:SUSE-cloud/barclamp-#{barclamp}.git")
+      end
+    end
+  end
+end
+
+# this task is only needed when we replace pull_upstream with rebase_upstream
+# the git rubygem offers no rebase method as of now
+desc "Update remotes."
+task :update_remotes => [:configure] do
+  @all_barclamps.each do |barclamp|
+    Dir.chdir File.join("barclamps", barclamp) do
+      git = Git::Lib.new
+      %w(upstream suse-cloud).each do |remote|
+        puts "Updating #{remote} of #{barclamp}..."
+        if git.remotes.include?(remote)
+          msg = git.fetch(remote, {})
+          puts msg unless msg.empty?
+        end
+      end
+    end
+  end
+end
+
+desc "Pull from upstream/master."
+task :pull_upstream => [:configure] do
+  @all_barclamps.each do |barclamp|
+    Dir.chdir File.join("barclamps", barclamp) do
+      git = Git::Lib.new
+      current_branch = git.branch_current
+      stash = { saved: false, branch: current_branch }
+
+      # save a stash if there are changes
+      unless git.diff_files.empty?
+        puts "Saving stash of #{barclamp}..."
+        git.stash_save("Automatic stash by rake - #{Time.now}")
+        stash[:saved] = true
+      end
+
+      # checkout master if on a different branch
+      unless current_branch == "master"
+        puts "Checking out master of #{barclamp}..."
+        git.checkout("master")
+      end
+
+      # pull changes from upstream/master
+      puts "Pull from upstream/master of #{barclamp}"
```

Comment:
This ```puts``` should be within the condition block, and i would print out another message if the remote does not exist.